### PR TITLE
feat: Double click selects word, Triple click selects line

### PR DESCRIPTION
**Pull request by nik-rev**
this PR improves mouse support by implementing 2 functionalities common in other text editors:
- Double-clicking selects the word
- Triple-clicking selects the full line

It is desirable to implement this in the editor as it provides cross-terminal support as well as, allowing functionality that terminal emulators won't be able to provide: such selecting the entire line in the document.

Would close https://github.com/helix-editor/helix/issues/12500

### CODE REVIEW

File: helix-view/src/input.rs [hunk 1]
```diff
@@ -59,6 +59,81 @@ pub enum MouseButton {
     /// Middle mouse button.
     Middle,
 }
+
+/// Tracks the character positions where the mouse was last clicked
+///
+/// If we double click, select that word
+/// If we triple click, select full line
```
Comment (lens0021) [inline helix-view/src/input.rs:66]: Should these comments be here, though they describe the actions? It seems that only the definition of what is the input should be here.

Comment (nik-rev) [reply helix-view/src/input.rs:66]: Yeah true, i've removed them

**Comment by nik-rev** [comment]
Just did some refactors, fixed a bug and added tests which consider various edge cases. PR is really ready for review now!

**Comment by nik-rev** [comment]
@kirawi could you also add waiting on review please?

**Comment by klaussilveira** [comment]
Would love to see this in some release. :eyes:

**Comment by nik-rev** [comment]
Closing! Reason: [State of my Helix PRs](https://github.com/helix-editor/helix/discussions/15171)
